libs/server/API/IGarnetAdvancedApi.cs

```diff
@@ -22,6 +22,14 @@ public interface IGarnetAdvancedApi
         /// <param name="wait"></param>
         bool GET_CompletePending((GarnetStatus, SpanByteAndMemory)[] outputArr, bool wait = false);
 
+        /// <summary>
+        /// Complete pending read operations on main store
+        /// </summary>
+        /// <param name="completedOutputs"></param>
+        /// <param name="wait"></param>
+        /// <returns></returns>
+        bool GET_CompletePending(out CompletedOutputIterator<SpanByte, SpanByte, SpanByte, SpanByteAndMemory, long> completedOutputs, bool wait = false);
```

Comment:
is it expected that *only* garnet will implement this interface? adding methods is a break, technically (and yes, I'm in no position to get fussy about this, since SE.Redis does this routinely - my point is "as long as this is considered and deliberate")

Reply:
It is mostly Garnet-only at this point, except for stored procedures, where we are giving users access to the same API to invoke in the procedure. At some point we want to separate out Garnet's advanced internal API from what we give users access to in the stored procedure, but they are the same today:

https://github.com/microsoft/garnet/blob/51667fd94e18b23a5d1b9212fb7571a1f97db4e1/libs/server/Custom/CustomTransactionProcedure.cs#L95-L96